hypervisor/src/mshv/mod.rs

```diff
@@ -395,6 +395,19 @@ impl hypervisor::Hypervisor for MshvHypervisor {
     }
 }
 
+#[cfg(feature = "sev_snp")]
+struct Ghcb(*mut svm_ghcb_base);
+
+#[cfg(feature = "sev_snp")]
+// SAFETY: struct is based on GHCB page in the hypervisor,
+// safe to Send across threads]
+unsafe impl Send for Ghcb {}
+
+#[cfg(feature = "sev_snp")]
+// SAFETY: struct is based on GHCB page in the hypervisor,
+// safe to Sync across threads
+unsafe impl Sync for Ghcb {}
```

Comment:
Do we need both `Send` and `Sync`? I thought `Send` was sufficient here.
It's not really true that it's safe to sync across threads without a mutex, is it?

Reply:
Yes. This is because of the Vcpu trait.
error[E0277]: `*mut mshv_bindings::svm_ghcb_base` cannot be shared between threads safely
   --> hypervisor/src/mshv/mod.rs:437:20
    |
437 | impl cpu::Vcpu for MshvVcpu {
    |                    ^^^^^^^^ `*mut mshv_bindings::svm_ghcb_base` cannot be shared between threads safely
    |
    = help: within `mshv::MshvVcpu`, the trait `std::marker::Sync` is not implemented for `*mut mshv_bindings::svm_ghcb_base`, which is required by `mshv::MshvVcpu: std::marker::Sync`
note: required because it appears within the type `mshv::Ghcb`
   --> hypervisor/src/mshv/mod.rs:399:8
    |
399 | struct Ghcb(*mut svm_ghcb_base);
    |        ^^^^
note: required because it appears within the type `std::option::Option<mshv::Ghcb>`
   --> /home/muislam/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:571:10
    |
571 | pub enum Option<T> {
    |          ^^^^^^
note: required because it appears within the type `mshv::MshvVcpu`
   --> hypervisor/src/mshv/mod.rs:412:12
    |
412 | pub struct MshvVcpu {
    |            ^^^^^^^^
note: required by a bound in `cpu::Vcpu`
   --> hypervisor/src/cpu.rs:351:24
    |
351 | pub trait Vcpu: Send + Sync {
    |                        ^^^^ required by this bound in `Vcpu`

Reply:
ok then!